dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1245,34 +1280,41 @@ class DXRIndexAction : public PluginASTAction {
       D.Report(DiagID) << output;
       return false;
     }
-    output = realpath(output.c_str(), NULL);
+    output = abs_output;
     output += "/";
+    FileInfo::output = output;
 
-    const char* tmp = getenv("DXR_CXX_CLANG_TEMP_FOLDER");
-    if(tmp){
+    // The temp directory for this plugin's output.
+    const char *tmp = getenv("DXR_CXX_CLANG_TEMP_FOLDER");
+    std::string tmpdir;
+    if (tmp)
       tmpdir = tmp;
-    }else
+    else
       tmpdir = output;
-
-    char* abs_tmpdir = realpath(tmpdir.c_str(), NULL);
-    if(!abs_tmpdir){
+    char *abs_tmpdir = realpath(tmpdir.c_str(), nullptr);
+    if (!abs_tmpdir) {
       DiagnosticsEngine &D = CI.getDiagnostics();
       unsigned DiagID = D.getCustomDiagID(DiagnosticsEngine::Error,
         "Temporary directory '%0' does not exist");
       D.Report(DiagID) << tmpdir;
       return false;
     }
-    tmpdir = realpath(tmpdir.c_str(), NULL);
+    tmpdir = abs_tmpdir;
     tmpdir += "/";
+    IndexConsumer::setTmpDir(tmpdir);
+
+    free(abs_src);
```

Comment:
Or we could stick that memory in a unique_ptr with a custom deleter and let it worry about when to delete:
`std::unique_ptr<char, decltype(&free)> abs_src(realpath(args[0].c_str(), nullptr), free);`
Yeah, I like that better (maybe with a typedef for that long type).  The concern I have with freeing as early as possible is that someone could edit later to use the freed var after the free, which is undefined behavior that may or may not get caught in testing with a given compiler.  Unlikely I think, but not impossible I guess.